src/java.base/share/classes/java/text/ChoiceFormat.java

```diff
@@ -172,20 +119,51 @@
  *
  * <dl>
  * <dt><i>Format:</i>
- * <dd>Any characters except the <i>Relation</i> symbols
+ * <dd>Any characters except the special pattern character '|'
  * </dl>
  *
  * </blockquote>
  *
  * <i>Note:The relation &le; is not equivalent to &lt;&equals;</i>
  *
- * <p>If a <i>Relation</i> symbol is to be used within a <i>Format</i> pattern,
- * it must be single quoted. For example,
- * {@code new ChoiceFormat("1# '#'1 ").format(1)} returns {@code " #1 "}.
- * Use two single quotes in a row to produce a literal single quote. For example,
```

Comment:
It sounds right to remove this restriction, but if we were to use `|` inside `Format`, shouldn't it still need to be single-quoted?

Reply:
Right the restriction shouldn't be removed, but rather updated. Fixed in https://github.com/openjdk/jdk/pull/17640/commits/740686b2052a276b77529bea5444692e5bbe0078